aws/table_aws_ec2_ami.go

```diff
@@ -258,6 +258,17 @@ func listEc2Amis(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData
 
 	input := &ec2.DescribeImagesInput{}
 
+	input.IncludeDisabled = new(bool)
+	disabledState := "disabled"
+	state := getQualsValueByColumn(d.Quals, "state", "string")
+	if val, ok := state.(string); ok {
+		*input.IncludeDisabled = val == disabledState
+	} else if val, ok := state.([]string); ok {
+		for _, v := range val {
+			*input.IncludeDisabled = v == disabledState
+		}
+	}
```

Comment:
```suggestion
	input := &ec2.DescribeImagesInput{
		IncludeDisabled: aws.Bool(false), // Default: No disabled AMIs are included in the response.
	}
	disabledState := "disabled"
	if d.EqualsQuals["state"] != nil { // Check for invalid memory address or nil pointer dereference error
		state := getQualsValueByColumn(d.Quals, "state", "string")
		if val, ok := state.(string); ok {
			*input.IncludeDisabled = val == disabledState
		} else if val, ok := state.([]string); ok {
			for _, v := range val {
				*input.IncludeDisabled = v == disabledState
			}
		}
	}
```

@razin99, could you please review the suggestions above? 

These suggestions were provided for the following reasons:
- The default API behavior excludes disabled AMIs from the response.
- With the current changes in the PR, the query `select name, state from aws_ec2_ami` is failing with the error `Error: rpc error: code = Internal desc = aws: rpc error: code = Internal desc = hydrate function listEc2Amis failed with panic runtime error: invalid memory address or nil pointer dereference (SQLSTATE HV000)`.